src/esrs/postgres/mod.rs

```diff
@@ -251,62 +189,51 @@ impl<
 }
 
 impl<
-        Evt: Serialize + DeserializeOwned + Send + Sync,
-        Err: From<sqlx::Error> + From<serde_json::Error> + Send + Sync,
-        Projector: PgProjector<Evt, Err> + Send + Sync + ?Sized,
-        Policy: PgPolicy<Evt, Err> + Send + Sync + ?Sized,
-    > ProjectorStore<Evt, PoolConnection<Postgres>, Err> for InnerPgStore<Evt, Err, Projector, Policy>
+        Event: Serialize + DeserializeOwned + Send + Sync,
+        Error: From<sqlx::Error> + From<serde_json::Error> + Send + Sync,
+        Projector: PgProjector<Event, Error> + Send + Sync + ?Sized,
+        Policy: PgPolicy<Event, Error> + Send + Sync + ?Sized,
+    > ProjectorStore<Event, Transaction<'_, Postgres>, Error> for InnerPgStore<Event, Error, Projector, Policy>
 {
     fn project_event<'a>(
         &'a self,
-        store_event: &'a StoreEvent<Evt>,
-        executor: &'a mut PoolConnection<Postgres>,
-    ) -> Pin<Box<dyn Future<Output = Result<(), Err>> + Send + 'a>>
+        store_event: &'a StoreEvent<Event>,
+        transaction: &'a mut Transaction<Postgres>,
+    ) -> Pin<Box<dyn Future<Output = Result<(), Error>> + Send + 'a>>
```

Comment:
Return type as `Pin<Box<dyn Future<Output = Result<(), Error>> + Send + 'a>>` is what exactly the `async_trait` crate does for traits. 
We can simplify it by using it.
I would guess that by using it we can remove the `'a` lifetime and the `Self: Sync` bound.